app/src/org/commcare/views/widgets/CommCareAudioWidget.java

```diff
@@ -105,88 +114,180 @@ public IAnswerData getAnswer() {
         player.pause();
         mPlayButton.setBackgroundResource(R.drawable.play);
         mPlayButton.setOnClickListener(v -> resumeAudioPlayer());
+        stopPlaybackTimer();
     }
 
     private void resumeAudioPlayer() {
         player.start();
         mPlayButton.setBackgroundResource(R.drawable.pause);
         mPlayButton.setOnClickListener(v -> pauseAudioPlayer());
+        startPlaybackTimer();
     }
 
     private void resetAudioPlayer() {
-        player.release();
-        mPlayButton.setBackgroundResource(R.drawable.play);
-        mPlayButton.setOnClickListener(v -> playAudio());
+        if (player != null) {
+            player.release();
+        }
+        stopPlaybackTimer();
     }
 
     @Override
     protected void togglePlayButton(boolean enabled) {
         if (enabled) {
-            mPlayButton.setBackgroundResource(R.drawable.play);
+            initAudioPlayer();
         } else {
-            mPlayButton.setBackgroundResource(R.drawable.play_disabled);
+            resetAudioPlayer();
+            hidePlaybackIndicators();
         }
-        mPlayButton.setEnabled(enabled);
+    }
+
+    private void hidePlaybackIndicators() {
+        mPlayButton.setVisibility(INVISIBLE);
+        playbackSeekBar.setVisibility(INVISIBLE);
+        playbackDuration.setVisibility(INVISIBLE);
+        playbackTime.setVisibility(INVISIBLE);
+    }
+
+    private void initAudioPlayer() {
+        mPlayButton.setVisibility(VISIBLE);
+        mPlayButton.setBackgroundResource(R.drawable.play);
+        mPlayButton.setOnClickListener(v -> playAudio());
+
+        Uri filePath = Uri.parse(mInstanceFolder + mBinaryName);
+        player = MediaPlayer.create(getContext(), filePath);
+        player.setOnCompletionListener(mp -> onCompletePlayback());
+
+        playbackDuration.setVisibility(VISIBLE);
+        playbackDuration.setText("/" + getTimeString(player.getDuration()));
+
+        playbackTime.setVisibility(VISIBLE);
+        playbackTime.setText("00:00");
+
+        playbackSeekBar.setVisibility(VISIBLE);
+        playbackSeekBar.setMax(player.getDuration() / 1000);
+        playbackSeekBar.setProgress(0);
+        playbackSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+
+            @Override
+            public void onStopTrackingTouch(SeekBar seekBar) {
+            }
+
+            @Override
+            public void onStartTrackingTouch(SeekBar seekBar) {
+            }
+
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+                if (player != null && fromUser) {
+                    player.seekTo(progress * 1000);
+                    playbackTime.setText(getTimeString(player.getCurrentPosition()));
+                }
+            }
+        });
+    }
+
+    private void onCompletePlayback() {
+        playbackSeekBar.setProgress(100);
+        stopPlaybackTimer();
+        playbackTime.setText("00:00");
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dimagi/commcare-android/blob/2aa03aecc11ade0381c1f426b782e030115c4729/app/src/org/commcare/views/widgets/CommCareAudioWidget.java#L282"></span>String literal in <code>setText</code> can not be translated. Use Android resources instead.</td>
    </tr>
  </tbody>
</table>